tls_manager.go

```diff
@@ -35,7 +35,10 @@ const (
 
 var (
 	// privateKeyPrefix is the prefix to a plaintext TLS key.
-	privateKeyPrefix = []byte("-----BEGIN EC PRIVATE KEY-----")
+	// It should match these two key formats:
+	// - `-----BEGIN PRIVATE KEY-----`    (PKCS8).
+	// - `-----BEGIN EC PRIVATE KEY-----` (SEC1/rfc5915, the legacy format).
+	privateKeyPrefix = []byte("-----BEGIN ")
```

Comment:
> To make this more robust and handle some rare edge cases that were previously covered by the prefix check, we might want to use a regexp or a double string match instead.

Agree this should only be kept temporarily. Perhaps this loose prefix check could cause issues if kept long term